src/app/CASESessionManager.h

```diff
@@ -58,11 +61,22 @@ class CASESessionManager : public SessionReleaseDelegate, public Dnssd::Resolver
 
         mConfig = params;
 
-        // TODO: Revisit who should be set as the resolver delegate
-        Dnssd::Resolver::Instance().SetResolverDelegate(this);
+        if (mConfig.dnsResolver == nullptr)
+        {
+#if CHIP_DEVICE_CONFIG_ENABLE_DNSSD
```

Comment:
For devices we should always require DNSSD to be avaialbe. If this not being the case we may use error macro to fail early.